### DISCUSSION TITLE
CSV Import?

### DISCUSSION THREAD

**Discussion by gassi101** [Tastaturberuf/anyStores#95] [Feature Requests / Funktionswünsche]
Hi,
ich müsste  die Daten via CSV importieren/updaten können?! Gibts dafür einen Ansatz ? Das alte Modul hatte diese Funktionalität ....
Tabimporter evtl.?!
Danke und Gruss
Gassi

**Comment by Tastaturberuf** [comment]
Im Moment am einfachsten über Adminer oder PhpMyAdmin. Die Import-Funktion müsste erstmal komplett überarbeitet werden. Die Nachfrage auch eher gering und deswegen ganz niedrige Prio.

**Comment by sfx12** [comment]
Die funktion würde ich auch gerne wieder haben !

**Comment by gassi101** [comment]
Also - ich finde es ist eigentlich sogar eine der wichtigsten Funktionen. Die meisten die das Modul anwenden - wollen dies via Excell füttern - und jeder der mehr wie 20 Standorte hat - wünscht sich die Funktion. Bei mir ginge es um ca. 9500 Einträge ;-)
Das möchte keiner von Hand eingeben!

**Comment by Tastaturberuf** [comment]
Wie ihr seht, seid ihr die ersten Beiden die das anfragen. Momentan mit PMA
oder Adminer kein Problem.

**Comment by gassi101** [comment]
Hey,
ich habe das importCSV tool von contao verwendet - das geht halbwegs - jedoch ohne Geokoordinaten. Problem ist - das man jedes mal halt alle Löschen muss und neu importieren -kein update möglich.... zudem danach immer alle nochmal bearbeiten muss um an die coords zu kommen... Aber ja - wenn das so selten angefragt wird - heisst es halt abwarten ;-) Ich hoffe auf jeden Fall - das es irgendwann mal wieder eingebaut wird....

**Comment by Tastaturberuf** [comment]
Die Coords kann man ja dann bequem mit **Alle Bearbeiten** -> **Überschreiben** -> **Lon, Lat leer** **Speichern** generieren lassen. 

Wenn du das dann exportierst musst ja künftig schon mal nicht mehr die Coords neu berechnen lassen.

**Comment by elcreativo** [comment]
@gassi101 Welches importCSV Tool von Contao hast du verwendet? Ich würde dieses Tool selbst gerne nutzen. Danke!

**Comment by gassi101** [comment]
hey elcreativo,
das hier:
https://contao.org/de/erweiterungsliste/view/import_from_csv.10000019.de.html  oder hier
https://github.com/markocupic/import-from-csv-bundle
an sich ok - nur man muss halt beim update immer alle vorher löschen... und alle neu hinzufügen.
Es gäbe wohl sogar ne hook vor die geo daten.... schau mal auf github diesbezüglich.

gruss Gassi

**Comment by overheaddesign** [comment]
Die Import Funktion würde bei mir auch gut Verwendung finden

**Comment by DanMan** [comment]
Also wenn es wirklich nur um das Füllen der `tl_anystores` Tabelle geht, dann braucht es dafür keinen extra Importer. Eine Doku wie man eine CSV von Hand einspielt (Unterschied id <> pid ?) und wie man die Koordinaten generieren kann sollte es auch tun.

**Comment by ContaoUser** [comment]
Irgendwie stehe ich auf dem Schlauch. AnyStores macht auf mich einen guten Eindruck, die Einbindung im Frontend und die Bedienung im Backend sind selbsterklärend. Also großes Lob an den Programmierer!
ABER: der Import von Adressen will nicht gelingen. Egal, ob ich "import form csv" einsetze (die Tabelle tl_anystores wird angezeigt und ich kann einen Import darüber machen) oder via phpMyAdmin importiere - unter "Standorte/Kategorie" gibt es keinen Eintrag. 
Da ich rund 800 Adressen vorliegen habe, habe ich wenig Bock, dass alles per Hand einzutippen …
Wenn ich die vorherigen Einträge lesen, scheint es ja zu funktionieren, nur ich mache etwas falsch (was auch immer das ist).
Verwende übrigens Contao 3.5.6 und AnyStores 1.9.1 …
Danke für jeglichen Tipp!

**Comment by Tastaturberuf** [comment]
Denke daran bei den Standorten eine `pid` mit der Kategorie-ID zu vergeben. 

Am besten legst du einen Standort an und exportierst diesen um die korrekte Datenbank-Struktur zu haben.

**Comment by ContaoUser** [comment]
Danke für die schnelle Antwort!
Dass mit der "Vorlage" als Muster-Standort hatte ich schon ausprobiert, aber die Kategorie-ID nicht bedacht. DANKE! Jetzt bin ich wieder ein gutes Stück weiter und die Standorte werden alle im Backend gelistet … aber ich bekomme die Aktualisierung der Geo-Daten nicht geregelt. Der Hook bei import_from_csv funzte nicht. Also doch "Handarbeit"

**Comment by Tastaturberuf** [comment]
Ein paar Antworten hoch scrollen. Überschreiben ist das Zauberwort.
Am 21.11.2016 8:57 nachm. schrieb ContaoUser notifications@github.com:Danke für die schnelle Antwort!
Dass mit der "Vorlage" als Muster-Standort hatte ich schon ausprobiert, aber die Kategorie-ID nicht bedacht. DANKE! Jetzt bin ich wieder ein gutes Stück weiter und die Standorte werden alle im Backend gelistet … aber ich bekomme die Aktualisierung der Geo-Daten nicht geregelt. Der Hook bei import_from_csv funzte nicht. Also doch "Handarbeit"

**Comment by ContaoUser** [comment]
DANKE! Genauso gehts! 
Jetzt funktioniert auch die Suchfunktion im Frontend.
